server/routes/comments.js

```diff
@@ -105,63 +95,98 @@ router.put(
 				.json({ msg: 'Comment not found' });
 		}
 
-		if (like !== undefined) {
-			if (comment.likes.includes(like)) {
-				res.json({ msg: 'You already like this comment' });
-			}
-
-			if (!ObjectId.isValid(like)) {
-				res.json({
-					msg: 'User Id is not valid',
-				});
-			}
-
-			comment = await Comment.findByIdAndUpdate(
-				req.params.id,
-				{ $push: { likes: like } },
-				{ new: true }
-			);
-
-			res.json(comment);
-		} else if (bodytext !== undefined) {
-			if (comment.author.toString() !== req.user.id) {
-				return res
-					.status(401)
-					.json({ msg: 'Not authorized' });
-			}
-
-			comment = await Comment.findByIdAndUpdate(
-				req.params.id,
-				{ bodytext },
-				{ new: true }
-			);
-
-			res.json(comment);
-		} else {
-			res.json({ msg: 'Comment update failed' });
+		if (comment.authorId.toString() !== req.user.id) {
+			return res
+				.status(401)
+				.json({ msg: 'Not authorized' });
 		}
+
+		comment = await Comment.findByIdAndUpdate(
+			req.params.id,
+			{ bodytext },
+			{ new: true }
+		);
+
+		res.json(comment);
 	})
 );
 
-// @route   DELETE api/comments/:id
-// @desc    Delete comment
+// @route   POST api/comments/:id/likes
+// @desc    Add like to comment
 // @access  Private
+router.post(
+	'/:id/likes',
+	auth,
+	errorCatcher(async (req, res) => {
+		let comment = await Comment.findById(req.params.id);
+
+		if (!comment) {
+			return res
+				.status(404)
+				.json({ msg: 'Comment not found' });
+		}
+
+		if (comment.likes.includes(ObjectId(req.user.id))) {
+			res.json({ msg: 'You already like this comment' });
+		}
```

Comment:
Were we wanting to `return` here so we don't execute the rest of the methods code?